packages/block-editor/src/components/off-canvas-editor/appender.js

```diff
@@ -27,18 +31,70 @@ export const Appender = forwardRef( ( props, ref ) => {
 		};
 	}, [] );
 
+	const { insertedBlockAttributes } = useSelect(
+		( select ) => {
+			const { getBlockAttributes } = select( blockEditorStore );
+
+			return {
+				insertedBlockAttributes: getBlockAttributes( insertedBlock ),
+			};
+		},
+		[ insertedBlock ]
+	);
```

Comment:
When the inserted block ID changes we need to refetch info about the block attributes so we can update the Link UI.